packages/ws-response/index.js

```diff
@@ -0,0 +1,62 @@
+import ApiGatewayManagementApi from 'aws-sdk/clients/apigatewaymanagementapi.js'// v2
+// import {ApiGatewayManagementApi} from '@aws-sdk/client-apigatewaymanagementapi' // v3
+
+import { canPrefetch, createClient, createPrefetchClient } from '@middy/util'
+
+const defaults = {
+  AwsClient: ApiGatewayManagementApi,
+  awsClientOptions: {}, // { endpoint }
+  awsClientAssumeRole: undefined,
+  awsClientCapture: undefined,
+  disablePrefetch: false
+}
+
+const wsResponderMiddleware = (opts) => {
+  const options = { ...defaults, ...opts }
+
+  let client
+  if (canPrefetch(options) && options.awsClientOptions.endpoint) {
+    client = createPrefetchClient(options)
+  }
+
+  const wsResponderMiddlewareAfter = async (request) => {
+    normalizeWsResponse(request)
+    const { response } = request
+
+    if (!response.ConnectionId) return
+
+    if (!options.awsClientOptions.endpoint && request.event.requestContext) {
+      options.awsClientOptions.endpoint = request.event.requestContext.domainName + '/' + request.event.requestContext.stage
+    }
+    if (!client) {
+      client = await createClient(options, request)
+    }
+
+    await client
+      .postToConnection(response)
+      .promise()
+
+    request.response = {
+      statusCode: 200
+    }
```

Comment:
It would be a good idea if we could somehow attach the original response to request if some other middleware wants it.
For example, in our use case, after having sent a response, we invoke the next step in the step function where we need the response and therefore cannot use this since it will change the response after having sent it.

Reply:
If you can create a PR to remove L39-41 and update any tests, that works for me.

Reply:
Will do shortly